assets/js/modules/analytics/util/index.js

```diff
@@ -131,15 +131,16 @@ export function extractAnalyticsDataForPieChart( reports, options = {} ) {
  *
  * @since 1.0.0
  *
- * @param {Array} rows          An array of rows to reduce.
- * @param {Array} selectedStats The currently selected stat we need to return data for.
+ * @param {Array}  rows           An array of rows to reduce.
+ * @param {number} selectedMetric The currently selected metric index.
+ * @param {number} selectedStats  The currently selected stat we need to return data for.
  * @return {Array} Array of selected stats from analytics row data.
  */
-function reduceAnalyticsRowsData( rows, selectedStats ) {
+function reduceAnalyticsRowsData( rows, selectedMetric, selectedStats ) {
```

Comment:
Potentially this is worth refining as a more reusable function - but if we touch this legacy function to improve it, we should think it through, i.e. not just patch what's needed but also try to make it more understandable. The naming here is completely unclear to me.

I suggest we either change this function further, or we introduce an updated (separate) version of it that new code should use. Either is fine with me, but we should make sure the new code only uses functions we can consider well reusable and understandable.

Reply:
This is a private function that is used by the `extractAnalyticsDashboardData` function only. I don't think it is worth reworking this function. I have just adopted it to work with a slightly different dataset than is used in the legacy component.